core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -1814,8 +1824,21 @@ public RealmConfiguration getRealmConfiguration() {
         String[] names = null;
         String domain = realmConfig
                 .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
-
-        String sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_ROLE_LIST);
+        String sqlStmt;
+        try (Connection dbConnection = getDBConnection()) {
+            if (dbConnection == null) {
+                throw new UserStoreException("null connection");
+            }
+             sqlStmt = isH2DB(dbConnection) ? realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_ROLE_LIST_H2) :
```

Comment:
check whether this line didn't exceed 120 characters

Reply:
The character count is 117 and is within the limit.